chirho/observational/handlers/soft_conditioning.py

```diff
@@ -288,12 +235,15 @@ def configure(
             torch.int32,
             torch.int64,
         ):
-            alpha = self.alpha * functools.reduce(
-                operator.mul, msg["fn"].event_shape, 1.0
-            )
-            return KernelSoftConditionReparam(
-                SoftEqKernel(alpha=alpha, event_dim=len(msg["fn"].event_shape))
-            )
+            support = constraints.boolean
```

Comment:
As above, you'll need to account for event dimensions here. I think you can just delete this conditional and set `support = msg["fn"].base_dist.support`.